server/src/test/java/org/elasticsearch/gateway/GatewayServiceTests.java

```diff
@@ -388,16 +387,14 @@ public void testScheduledRecoveryWithRecoverAfterNodes() {
             elapsed.millis(),
             () -> setDataNodeCountTaskQueue.submitTask(randomAlphaOfLength(5), new SetDataNodeCountTask(recoverAfterNodes - 1), null)
         );
-        deterministicTaskQueue.advanceTime();
-        deterministicTaskQueue.runAllRunnableTasks();
+        deterministicTaskQueue.runTasksUpToTimeInOrder(elapsed.millis());
 
         // The 2nd scheduled recovery when data nodes are above recoverAfterDataNodes again
         deterministicTaskQueue.scheduleAt(
             elapsed.millis() * 2,
             () -> setDataNodeCountTaskQueue.submitTask(randomAlphaOfLength(5), new SetDataNodeCountTask(recoverAfterNodes), null)
         );
-        deterministicTaskQueue.advanceTime();
-        deterministicTaskQueue.runAllRunnableTasks();
+        deterministicTaskQueue.runTasksUpToTimeInOrder(elapsed.millis() * 2);
```

Comment:
This usage seems to suggest a `scheduleAtAndRunUpTo` method. Let me know if you think so as well. Happy to iterate.

Reply:
I'm not sure it's that common a pattern elsewhere, but that still seems like a reasonable thing to add indeed.

Reply:
Great. I added the method and used it here and in two other places.